crates/fs/private/hamt/node.rs

```diff
@@ -43,20 +46,20 @@ where
 
 impl<K, V, H> Node<K, V, H>
 where
-    K: DeserializeOwned + Serialize + AsRef<[u8]> + Clone + Eq + PartialOrd + Debug,
+    K: DeserializeOwned + Serialize + Clone + Debug + AsRef<[u8]>,
     V: DeserializeOwned + Serialize + Clone + Debug,
     H: Hasher + Clone + Debug,
 {
     /// Sets a new value at the given key.
     pub async fn set<B: BlockStore>(
-        self: Rc<Self>,
+        self: &Rc<Self>,
```

Comment:
Me learning rust: 
I'm now noticing that in some places we use `&Rc<X>`.
Shouldn't we use `Rc<X>` instead? I kind of feel like `Rc` is already like `&`, and `&Rc<X>` to me looks like doing `&&X`, which would be weird.
One problem with using `Rc<X>` is sometimes you want to have a bound on the lifetime of some return value to be defined by the lifetime of that `Rc`, but what is the `Rc`-equivalent syntax for this: `&'a X`?

Reply:
Yes. It should be `Rc` unless there is a lifetime req like you mentioned

Reply:
> but what is the `Rc`-equivalent syntax for this: `&'a X`?
There is none. Rc is a runtime concept. Lifetimes are a compile-time concept. They don't work together.